.github/workflows/cloc-repository.yaml

```diff
@@ -0,0 +1,13 @@
+name: Count Lines of Code of This Repository
+
+on:
+  workflow_call:
+
+jobs:
+  cloc-repository:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          export FORMAT=default
+          ./scripts/cloc-repository.sh
```

Comment:
How necessary is this workflow?
If the repo is onboarded onto sonarcloud then it will automatically give you lines of code
Although I suppose if we want this to be a generic template we should have provider independent ways of getting lines of code.
But where is this data stored?

Reply:
The intention for this action is to be moved to the "deployment" or "release" stage which then enables us to implement fitness functions to examine delta changes, hotspots of the code and contribute to the analysis of some structural characteristics of the codebase (yes, there is more to come!). The other alternative is to run it on demand. The value of this is the ability to programmatically access the information. I would like to put it on trial with all the repos we have in the NHS England Tools org to see how practical and beneficial it is. Then we can either extend it or bin it. I'll be happy to show what other code metrics I have in mind for this trial...